### DISCUSSION TITLE
PWM fan control

### DISCUSSION THREAD

**Discussion by ogg1e** [mebs/CM4-NAS#37] [General]
Is there a fan control script that you have that you could share?
I have the same temp sensor and a noctua pwm fan. But I only see guides out there for using the cpu temp to control a fan.

**Comment by ogg1e** [comment]
I was able to see the temp reading from the sensor by enabling the overlay and modprobe etc. Just need to figure out how to take that reading and control the fan.

**Comment by mebs** [comment]
The fan controller is an I²C device, you can look at its datasheet to see what each registry does, but here's the command to set the fan RPM:
`i2cset -y 10 0x2f 0x30 [VALUE]`
With value ranging from 0x00 (off) to 0xff (max speed).
I found that running the fan I have at 0x70 all the time was very silent and more than enough to cool everything, so I ditched the script a while back.

edit: the range for fan speed may be inverted, there's a registry in the controller you can set to invert it iirc.
